presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -495,9 +495,9 @@ public PlanNode visitProject(ProjectNode node, RewriteContext<Set<Symbol>> conte
             ImmutableSet.Builder<Symbol> expectedInputs = ImmutableSet.builder();
 
             Assignments.Builder builder = Assignments.builder();
-            for (int i = 0; i < node.getOutputSymbols().size(); i++) {
-                Symbol output = node.getOutputSymbols().get(i);
```

Comment:
This is doing `ImmutableList.copyOf(assignments.keySet())` for every call, so it is indeed expensive.